main.go

```diff
@@ -33,36 +35,19 @@ func (svc *Service) getProtectedResource(c *gin.Context) {
 			return
 		}
 
-		lnMacaroonReq := lnrpc.BakeMacaroonRequest{
-			Permissions: []*lnrpc.MacaroonPermission{
-				{
-					Entity: "invoices",
-					Action: "read",
-				},
-				{
-					Entity: "invoices",
-					Action: "write",
-				},
-				{
-					Entity: "address",
-					Action: "read",
-				},
-				{
-					Entity: "address",
-					Action: "write",
-				},
-			},
-			AllowExternalPermissions: false,
+		mac, err := macaroon.BakeMacaroon(lndInvoice.RHash)
+		if err != nil {
+			c.Error(err)
+			return
 		}
-
-		lndMacaroon, err := svc.lndClient.BakeMacaroon(ctx, &lnMacaroonReq)
+		macBytes, err := mac.MarshalBinary()
 		if err != nil {
 			c.Error(err)
 			return
 		}
 
-		macaroon := lndMacaroon.Macaroon
 		invoice := lndInvoice.PaymentRequest
+		macaroon := base64.StdEncoding.EncodeToString(macBytes)
```

Comment:
I'd move this directly in the macaroon package. 

then we have one part of the code that takes care of generating that macaroon and the correct encoding. so we get one string that we simply pass on in the header. 

This gives flexibility to later change the genration of the macaroon token without changing the rest here. 

maybe macaroon.getMacaroonAsString(lndInvoice.RHash) ?